include/boost/json/detail/parse_into.hpp

```diff
@@ -867,8 +867,8 @@ class converting_handler<tuple_conversion_tag, T, P>
                 F{ std::forward_as_tuple(ec, args...) } } ); \
     }
 
-    BOOST_JSON_HANDLE_EVENT( on_object_begin );
-    BOOST_JSON_HANDLE_EVENT( on_object_end );
+    BOOST_JSON_HANDLE_EVENT( on_object_begin )
```

Comment:
Rather than removing semicolon here and in other 100500 places, remove it in the macro definition, please.

Reply:
Ah.  The macro itself doesn't seem to be the problem.  It's ending with a function, which ought not end with a semicolon.

```
#define BOOST_JSON_HANDLE_EVENT(fn) \
    struct do_ ## fn \
    { \
        template< class H, class... Args > \
        bool operator()( H& h, Args& ... args ) const \
        { \
            return h. fn (args...); \
        } \
    }; \
       \
    template< class... Args > \
    bool fn( error_code& ec, Args&& ... args ) \
    { \
        if( inner_active_ < 0 ) \
        { \
            BOOST_JSON_FAIL( ec, error::not_array ); \
            return false; \
        } \
        constexpr int N = std::tuple_size<T>::value; \
        if( inner_active_ >= N ) \
        { \
            BOOST_JSON_FAIL( ec, error::size_mismatch ); \
            return false; \
        } \
        using F = handler_op_invoker< do_ ## fn, error_code, Args...>; \
        using H = decltype(handlers_); \
        return mp11::mp_with_index<N>( \
            inner_active_, \
            tuple_handler_op_invoker<H, F>{ \
                handlers_, \
                F{ std::forward_as_tuple(ec, args...) } } ); \
    }

    BOOST_JSON_HANDLE_EVENT( on_object_begin );
    BOOST_JSON_HANDLE_EVENT( on_object_end );

```

Reply:
Oh, sorry, I was confused.